### PR TITLE
ISPN-15008 add overview of config files in the server README

### PR DESCRIPTION
**Pull request by domiborges**
https://issues.redhat.com/browse/ISPN-15008

### CODE REVIEW

**Comment by dvagnero** [comment]
@karesti does it make sense to add the steps on how use the config files in the server doc? We did it for log4j but at the same time there are instructions inside the xml files.

File: server/runtime/src/main/server/README.adoc [hunk 1]
```diff
@@ -42,6 +42,14 @@ Server configuration extends ${infinispan.brand.name} configuration with the fol
 `socket-bindings`:: Maps endpoint connectors to interfaces and ports.
 
 The default configuration file is `<SERVER_HOME>/server/conf/infinispan.xml`.
+${infinispan.brand.name} provides other ready-to-use configuration files that are primarily for development and testing purposes.
+
+`<SERVER_HOME>/server/conf/` provides the following configuration files:
+
+`infinispan-dev-mode.xml`:: Configures ${infinispan.brand.name} Server specifically for cross-site replication with IP multicast discovery. The configuration provides `BASIC` authentication to connect to the Hot Rod and REST endpoints.
```
Comment (karesti) [inline server/runtime/src/main/server/README.adoc:49]: I would add explicitly that this is for dev mode only, not for production in any case

File: server/runtime/src/main/server/README.adoc [hunk 2]
```diff
@@ -42,6 +42,14 @@ Server configuration extends ${infinispan.brand.name} configuration with the fol
...
+`infinispan-local.xml`:: Configures ${infinispan.brand.name} Server without clustering capabilities.
+`infinispan-xsite.xml`:: Configures cross-site replication on a single host and uses IP multicast for discovery.
+`log4j2.xml`:: Configures ${infinispan.brand.name} Server logging.
```
Comment (karesti) [inline server/runtime/src/main/server/README.adoc:52]: aren't we missing the infinispan.xml ?

**Review by karesti — CHANGES_REQUESTED** [review]

File: server/runtime/src/main/server/README.adoc (hunk 2 above)
Comment (dvagnero) [reply server/runtime/src/main/server/README.adoc:55]: Good point

File: server/runtime/src/main/server/README.adoc (hunk 1 above)
Comment (dvagnero) [reply server/runtime/src/main/server/README.adoc]: Updated. Better be safe than sorry

**Pull request merged**